### PR TITLE
Feat: Add "discover addresses" using a permanode

### PR DESCRIPTION
**Pull request by msarcev**
## Summary

- Generate addresses not known by the sdk or locally and query the address against a permanode to know if it had history
- Persist all known addresses in localStorage to save requests in the future
- Use gap logic in the search algorithm to know when to stop generating addresses

Resolves https://github.com/iotaledger/firefly/issues/7594

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [x] MacOS
    -   [ ] Linux
    -   [ ] Windows

Network: Testnet

### CODE REVIEW

File: packages/desktop/components/popups/AddressHistoryPopup.svelte [hunk 1]
```diff
@@ -1,36 +1,156 @@
...
+        } catch (err) {
+            console.error(err)
+            error = 'Couldn\'t generate a new address'
+        }
+
+        return [nextUnknownAddress, searchAddressStartIndex - 1]
+    }
+
+    async function search(): Promise<void> {
+        currendSearchGap = 0
```
Comment (brancoder) [inline packages/desktop/components/popups/AddressHistoryPopup.svelte:93]: fix typo in variable name `currentSearchGap`

**Review by brancoder — CHANGES_REQUESTED** [review]
Fix type check and lint errors

File: packages/shared/lib/core/network/constants/chronicle-urls.constant.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { NetworkId } from '../enums'
+
+export const CHRONICLE_URLS: Readonly<{ [key in NetworkId]?: string[] }> = {
+    [NetworkId.Iota]: ['https://chronicle.stardust-mainnet.iotaledger.net/'],
```
Comment (begonaalvarezd) [inline packages/shared/lib/core/network/constants/chronicle-urls.constant.ts:4]: the rotating addresses most likely come from chrysalis not stardust, shouldnt we check the historical addresses in chrysalis instead?

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]

File: packages/desktop/components/popups/AddressHistoryPopup.svelte [hunk 3]
```diff
@@ -1,36 +1,156 @@
...
+                .catch((err) => {
+                    console.error(err)
+                })
+        }
+
+        if (CHRONICLE_URLS[network] && CHRONICLE_URLS[network].length > 0) {
+            const chronicleRoot = CHRONICLE_URLS[network][0]
+            searchURL = `${chronicleRoot}${CHRONICLE_ADDRESS_HISTORY_ROUTE}`
+        } else {
+            error = 'Chronicle not configured'
```
Comment (begonaalvarezd) [inline packages/desktop/components/popups/AddressHistoryPopup.svelte:57]: All error messages should be internationalized

File: packages/desktop/components/popups/AddressHistoryPopup.svelte [hunk 4]
```diff
@@ -1,36 +1,156 @@
...
+    })
+
+    async function isAddressWithHistory(address: string): Promise<boolean> {
+        try {
+            const response = await fetchWithTimeout(`${searchURL}${address}`, 3, { method: 'GET' })
+            const addressHistory: AddressHistory = await response.json()
+            return addressHistory?.items?.length > 0
+        } catch (err) {
+            console.error(err)
+            error = 'Couldn\'t fetch address history'
```
Comment (begonaalvarezd) [inline packages/desktop/components/popups/AddressHistoryPopup.svelte:68]: All error messages should be internationalized

File: packages/desktop/components/popups/AddressHistoryPopup.svelte [hunk 5]
```diff
@@ -1,36 +1,156 @@
...
+                nextUnknownAddress = await getProfileManager().generateEd25519Address(
+                    accountIndex,
+                    searchAddressStartIndex
+                )
+
+                searchAddressStartIndex++
+            } while (knownAddresses.map((accountAddress) => accountAddress.address).includes(nextUnknownAddress))
+        } catch (err) {
+            console.error(err)
+            error = 'Couldn\'t generate a new address'
```
Comment (begonaalvarezd) [inline packages/desktop/components/popups/AddressHistoryPopup.svelte:86]: All error messages should be internationalized

File: packages/shared/lib/core/network/constants/chronicle-urls.constant.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+    [NetworkId.IotaAlphanet]: ['https://chronicle.alphanet.iotaledger.net/'],
+    [NetworkId.Shimmer]: ['https://chronicle.shimmer.network/'],
+    [NetworkId.Testnet]: ['https://chronicle.testnet.shimmer.network/'],
```
Comment (begonaalvarezd) [inline packages/shared/lib/core/network/constants/chronicle-urls.constant.ts:7]: if we use chrysalis we can likely remove these

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]

**Comment by msarcev** [comment]
I had to modify `checkAndMigrateChrysalisProfiles` here for the new field in `IPersistedAccountData`. Please confirm that this is fine. :D

**Review by begonaalvarezd — APPROVED** [review]
🚀 works

**Review by brancoder — APPROVED** [review]
LGTM :+1:

**Pull request merged**